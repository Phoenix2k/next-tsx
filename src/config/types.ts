export interface Config {
  a11yEnv: Array<string>;
  pages: Pages;
  theme: Theme;
}

export interface Page {
  navLabel: string;
  slug: string;
  title: string;
  url: string;
}

export interface Pages {
  [key: string]: Page;
}

export interface Theme {
  classes: {
    navLinkActive: string;
  };
  colors: {
    backgroundColor: string;
    borderColor: string;
    color: string;
    navLink: string;
    navLinkActive: string;
  };
  typography: {
    sansSerif: string;
  };
}
