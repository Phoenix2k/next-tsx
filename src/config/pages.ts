import sitemap from '../../data/sitemap.json';
import { Pages } from './types';

const pages: Pages = sitemap;

export default pages;
