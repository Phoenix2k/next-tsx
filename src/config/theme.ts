import { Theme } from './types';

const theme: Theme = {
  classes: {
    navLinkActive: 'nav-link--active'
  },
  colors: {
    backgroundColor: 'rgb(255,255,255)',
    borderColor: '#ccc',
    color: 'black',
    navLink: 'blue',
    navLinkActive: 'rgb(0, 128, 0)'
  },
  typography: {
    sansSerif: '"Open Sans", Helvetica, Arial, sans-serif'
  }
};

export default theme;
