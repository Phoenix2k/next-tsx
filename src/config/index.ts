import pages from './pages';
import theme from './theme';
import { Config } from './types';

const config: Config = {
  a11yEnv: ['development'],
  pages,
  theme
};

export default config;
