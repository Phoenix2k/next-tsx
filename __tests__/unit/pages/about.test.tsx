/// <reference types="jest"/>

import { ThemeProvider } from 'emotion-theming';
import { shallow } from 'enzyme';
import { NextRouter } from 'next/router';
import Renderer from 'react-test-renderer';
import config from '../../../src/config';
import About from '../../../src/pages/about';

jest.mock('next/router', () => ({
  useRouter(): Partial<NextRouter> {
    return {
      route: '/about'
    };
  }
}));

describe('About page', () => {
  it('renders "About page" heading', () => {
    const shallowRender = shallow(<About />);
    expect(shallowRender.find('h1').text()).toEqual('About page');
  });
  it('matches snapshot', () => {
    const snapshotRender = Renderer.create(
      <ThemeProvider theme={config.theme}>
        <About />
      </ThemeProvider>
    );
    const tree = snapshotRender.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
