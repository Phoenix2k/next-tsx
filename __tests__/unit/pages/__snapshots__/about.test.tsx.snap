// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page matches snapshot 1`] = `
<div
  className="css-1qibdk7-LayoutContainer evfywwz0"
>
  <header>
    Header
  </header>
  <nav
    className="css-1el57au-NavContainer ekr6xpg0"
  >
    <ul>
      <li>
        <a
          className="link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
    </ul>
  </nav>
  <main>
    <h1
      itemProp="title"
    >
      About page
    </h1>
  </main>
  <footer>
    Footer
  </footer>
</div>
`;
